presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1228,6 +1217,38 @@ private ActualProperties derivePropertiesRecursively(PlanNode result)
         {
             return PropertyDerivations.derivePropertiesRecursively(result, metadata, session, types, parser);
         }
+
+        private ExchangeNode createPartitionedRemoteExchange(PlanNode child, List<Symbol> partitioningColumns, Optional<Symbol> hashColumn)
+        {
+            return createPartitionedRemoteExchange(child, partitioningColumns, hashColumn, false);
+        }
+
+        private ExchangeNode createPartitionedRemoteExchange(PlanNode child, List<Symbol> partitioningColumns, Optional<Symbol> hashColumn, boolean replicateNullsAndAny)
+        {
+            return partitionedExchange(
+                    idAllocator.getNextId(),
+                    REMOTE,
+                    child,
+                    new PartitioningScheme(
+                            createPartitioning(partitioningColumns),
+                            child.getOutputSymbols(),
+                            hashColumn,
+                            replicateNullsAndAny,
+                            Optional.empty()));
+        }
+
+        private Partitioning createPartitioning(List<Symbol> partitioningColumns)
+        {
+            if (SYSTEM_PARTITIONING_PROVIDER.equals(partitioningProvider)) {
```

Comment:
I am wondering if it makes sense to let `SystemTablesMetadata.getPartitioningHandle` to return this `Partitioning`? This way it's all unified (instead of cheating for `SystemPartitioning` :P ) 

We can add a TODO and left this refactor as future work, though :).

Reply:
That's a good idea. Let's leave it for a separate PR. I will add a TODO